### ISSUE TITLE
Enhancement: [no-duplicate-export] <a short description of my proposal>

### ISSUE THREAD

**Issue by spearmootz** [typescript-eslint/typescript-eslint#6194]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/

### Description

when converting from javascript to typescript there are certain things we do to keep compatibility and migrate a little at a time. one of those is to export in a particular way to not have to rewrite everything.   so i ran into the failure example.

export class SomeClass {}

and then below left a module.exports = { something }

one way to accomplish this is to allow the following methods to coexist

```
(although these should fail because they are the same name, but pass otherwise)
export const varName
module.exports.varName
```

OR

```
module.exports = {}
```

but these two should not be mixed.

### Fail

```typescript
export const x = 'test';

module.exports = 'test';
module.exports = 'test';
```

### Pass

```typescript
export const x = 'test';

module.exports.y = 'test
```

**Comment by omril1** [comment]
Is there anything specific to typescript in here?
Seems like anyone who migrates from `javascript in commonjs` to `javascript in esm` can experience the same issue

**Comment by spearmootz** [comment]
Not an expert on module types. I guess I've never used the other syntax in regular js. I guess if you, @omril1 , think it's not a typescript issue please close the ticket and I'll open it in the regular es lint rule library.

**Comment by omril1** [comment]
@spearmootz I'm not a maintainer so can't close it, I was just trying to help since I like this project.
Maybe wait for an official answer from a maintainer

**Comment by bradzacher** [comment]
Yup, I'd agree with @omril1 here.
There's nothing TS-specific here.
The fact that `module.exports` conflicts with es6 exports is an implementation detail of the build step.

Additionally this is a very niche proposal - there are going to be very few TS codebases actively migrating from commonjs to es6 modules.

For the above reasons this isn't a rule we'll be looking to implement in this project. Thanks for filing!

---

Personally I would look to just codemod your codebase so you can do the migration to es6 modules in one hit and then ban commonjs altogether, rather than trying to support having both at the same time.
Supporting both and interoping is complex and difficult to maintain.
